web_infrastructure/jenkins_plugin.py

```diff
@@ -0,0 +1,753 @@
+                details=str(e))
+
+        # Change owner and permissions of the file
+        try:
+            os.chmod(f, 0664)
+            os.chown(
+                f,
+                getpwnam(self.params['jenkins_user']).pw_uid,
+                getpwnam(self.params['jenkins_group']).pw_gid)
+        except OSError, e:
+            self.module.fail_json(
+                msg="Cannot change plugin file permissions.",
+                details=str(e))
+
+    def uninstall(self):
+        changed = False
+
+        # Perform the action
+        if self.is_installed:
+            if not self.module.check_mode:
+                self.__pm_query('doUninstall', 'Uninstallation')
+
+            changed = True
+
+        return changed
+
+    def pin(self):
+        return self.__pinning('pin')
+
+    def unpin(self):
+        return self.__pinning('unpin')
+
+    def __pinning(self, action):
+        changed = False
+
+        # Check if the plugin is pinned/unpinned
+        if (
+                action == 'pin' and not self.is_pinned or
+                action == 'unpin' and self.is_pinned):
+
+            # Perform the action
+            if not self.module.check_mode:
+                self.__pm_query(action, "%sning" % action.capitalize())
+
+            changed = True
+
+        return changed
+
+    def enable(self):
+        return self.__enabling('enable')
+
+    def disable(self):
+        return self.__enabling('disable')
+
+    def __enabling(self, action):
+        changed = False
+
+        # Check if the plugin is pinned/unpinned
+        if (
+                action == 'enable' and not self.is_enabled or
+                action == 'disable' and self.is_enabled):
+
+            # Perform the action
+            if not self.module.check_mode:
+                self.__pm_query(
+                    "make%sd" % action.capitalize(),
+                    "%sing" % action[:-1].capitalize())
+
+            changed = True
+
+        return changed
+
+    def __pm_query(self, action, msg):
+        url = "%s/pluginManager/plugin/%s/%s" % (
+            self.params['url'], self.params['name'], action)
+        payload = {}
+
+        try:
+            r, info = fetch_url(
+                self.module, url, data=json.dumps(payload))
+
+            if info['status'] != 200:
+                self.module.fail_json(
+                    msg="Plugin not found. %s" % url,
+                    details=info['msg'])
+        except Exception, e:
+            self.module.fail_json(
+                msg="%s has failed." % msg,
+                details=e.message)
+
+
+def main():
+    # Module settings
+    module = AnsibleModule(
+        argument_spec=dict(
+            jenkins_group=dict(default='jenkins'),
+            jenkins_home=dict(default='/var/lib/jenkins'),
+            jenkins_user=dict(default='jenkins'),
+            name=dict(required=True),
+            params=dict(),
+            state=dict(
+                choices=[
```

Comment:
Yes, this is exactly what I was saying. The version must be as part of the name of the package. There is no extra param for version. This module has an extra `version` param, this is why I believe the `latest` should be specified in that param instead of in the `state`.